### PR TITLE
fix(viz treemap): Cannot nest more than three levels

### PR DESCRIPTION
**Pull request by newtoncy**
### SUMMARY

Treemap cannot be nested more than three levels before I modify it.
If the nesting is more than three levels, keyerror will be raised.
The unit tests I added can replicate this problem.

### TEST PLAN

Create a new chart

Select tree map

Select at least three dimensions in group by, and each dimension has at least two cardinalities

Run this query

This will trigger a keyerror before I modify it.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/viz_tests.py
Comment (villebro) [inline tests/viz_tests.py:1433]: nit: comment not needed after the fix is merged.
```suggestion
```

**Review by villebro — APPROVED** [review]
Thanks for the fix! LGTM with small nit.

File: superset/viz.py
Comment (mistercrunch) [inline superset/viz.py:975]: Should this be a `set` or a `list`? Sets don't preserve ordering, without full context, it feels like ordering may matter here (?)

Comment (newtoncy) [reply superset/viz.py]: The order doesn't seem to matter, because the front end reorders by size.
But I thought it might be better to keep it in order, so I added a few lines

**Review by villebro — APPROVED** [review]

**Comment by junlincc** [comment]
Screen Shot 2020-11-19 at 10 22 57 AM
Thank you, Chaoyi @newtoncy for the fix. This is what I'm seeing nested 3 layers on tree map, we need to either truncate the group labels or not show when the grid is small, would you mind making the changes? 
consider Tableau's treatment below
Treemap-many-dim-members

Please don't merge yet until cosmetic issues are addressed🙏 @villebro @mistercrunch

**Pull request closed without merging**
